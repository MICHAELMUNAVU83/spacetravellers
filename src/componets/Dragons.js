import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchingDragonsApi, changeDragonReservation } from '../redux/dragons';

function Dragon() {
  const dragons = useSelector((state) => state.dragons);
  const dispatch = useDispatch();
  useEffect(() => {
    if (!dragons.length) dispatch(fetchingDragonsApi());
  }, [dispatch]);
  const changeReservation = (e) => {
    dispatch(changeDragonReservation(e.target.id));
  };
  const render = dragons.map((dragon) => (
    <div className="each-rocket" key={dragon.id}>
      <img src={dragon.flickr_image} alt="flick" />
      <div className="rocket-details">
        <h3>
          {' '}
          {dragon.dragon_name}
        </h3>
        <p className="description">
          {dragon.reserved && (
          <button type="button" className="btn btn-primary btn-sm">
            Reserved
          </button>
          )}
          {' '}
          {dragon.description}
        </p>
        {!dragon.reserved && (
        <button id={dragon.id} type="button" className="btn btn-primary btn-lg" onClick={changeReservation}>
          RESERVE
        </button>
        )}
        {dragon.reserved && (
        <button id={dragon.id} type="button" className="btn btn-outline-secondary" onClick={changeReservation}>
          CANCEL RESERVATION
        </button>
        )}

      </div>

    </div>
  ));
  return (
    <div>
      {render}
    </div>
  );
}

export default Dragon;
