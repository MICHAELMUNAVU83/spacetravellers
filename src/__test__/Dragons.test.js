import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Dragons from '../componets/Dragons';
import store from '../redux/configureStore';

it('Dragons renders correctly', () => {
  const tree = TestRenderer.create(
    <Provider store={store}>
      <Dragons />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
