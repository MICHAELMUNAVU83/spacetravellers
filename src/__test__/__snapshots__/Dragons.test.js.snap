// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Qoute renders correctly 1`] = `<div />`;
